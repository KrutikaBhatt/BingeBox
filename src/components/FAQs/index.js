import React, { useState, useContext, createContext } from 'react';
import { Container, Frame, Title, Item, Inner, Header, Body } from './styles/faq';

const ToggleContext = createContext();

export default function FAQs({ children, ...restProps }) {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
}

FAQs.Title = function FAQsTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

FAQs.Frame = function FAQsFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

FAQs.Item = function FAQsItem({ children, ...restProps }) {
  const [toggleShow, setToggleShow] = useState(false);

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

FAQs.Header = function FAQsHeader({ children, ...restProps }) {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header onClick={() => setToggleShow(!toggleShow)} {...restProps} >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

FAQs.Body = function FAQsBody({ children, ...restProps }) {
  const { toggleShow } = useContext(ToggleContext);

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};
