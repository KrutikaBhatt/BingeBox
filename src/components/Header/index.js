import React,{useState} from 'react';
import {Link as ReactRouterLink} from 'react-router-dom';
import {Background ,Container,Logo,ButtonLink,Feature,Text , FeatureCallOut,Link, Group, Dropdown,Picture,Profile,Search,SearchIcon,SearchInput,PlayButton,DropdownItemLink} from './styles/header';

export default function Header({bg =true,children,...restProps}){
    return bg ? <Background {...restProps}>{children}</Background> :children;
}

Header.Frame =function HeaderFrame({children,...restProps}){
    return(
        <Container {...restProps}>
            {children}
        </Container>
    );
};
Header.Callout = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
  };

  Header.Group = function HeaderGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>;
  };

Header.Logo  = function HeaderLogo({imageLink,...restProps}){
    return(
        <ReactRouterLink to={imageLink}>
            <Logo {...restProps}/>
        </ReactRouterLink>
    );
};


Header.SearchBar = function HeaderSearch({ searchTerm, setsearchTerm, ...restProps }) {
  const [searchActive, setsearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon onClick={() => setsearchActive((searchActive) => !searchActive)} data-testid="search-click">
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setsearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
        data-testid="search-input"
      />
    </Search>
  );
  
};
Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature>{children}</Feature>;
  };
  
Header.ButtonLink = function HeaderButton({children , ...restProps}){
    return (<ButtonLink {...restProps}>{children}</ButtonLink>);
};

Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
  };

  Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>;
  };
  Header.DropItem = function HeaderTextLink({ children, ...restProps }) {
    return <DropdownItemLink {...restProps}>{children}</DropdownItemLink>;
  };
  
Header.Picture = function HeaderPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />;
  };
  
  Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>;
  };
  
Header.Profile = function HeaderProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>;
  };

  
Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};