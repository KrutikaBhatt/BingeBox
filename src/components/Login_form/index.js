import React from 'react';

import { Container, Error, Base, Title, Text, TextSmall, Link, Input, Submit } from './styles/form';

export default function Login_form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Login_form.Error = function Login_formError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Login_form.Base = function Login_formBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Login_form.Title = function Login_formTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Login_form.Text = function Login_formText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Login_form.TextSmall = function Login_formTextSmall({ children, ...restProps }) {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Login_form.Link = function Login_formLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Login_form.Input = function Login_formInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Login_form.Submit = function Login_formSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};
