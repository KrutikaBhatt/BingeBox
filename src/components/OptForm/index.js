import React from 'react';
import {Container,Input,Text,Button,Break} from './styles/optForm';
export default function OptingForm({children, ...restProps}){
    return(
        <Container {...restProps}>
            {children}
        </Container>
    );
}

OptingForm.Input = function OptingFormInput({...restProps}){
    return <Input {...restProps}/>;
};

OptingForm.Button = function OptingFormButton({children,...restProps}){
    return (
        <Button {...restProps}>
            {children}
            <img src ="/images/icons/chevron-right.png" alt ="Subscribe now" />
        </Button>
    );
};

OptingForm.Text = function OptingFormText({children,...restProps}){
    return (
       <Text {...restProps}>
           {children}
       </Text>
    );
};

OptingForm.Break = function OptingFormBreak({...restProps}){
    return (
        <Break {...restProps}>
        </Break>
    );
};