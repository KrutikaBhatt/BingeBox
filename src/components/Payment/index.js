import React from 'react';
import {Container,Title,Button,Col,SubTitle,Emp,Price_box,Head} from './styles/payment';

export default function Payment({children,...restProps}){
    return <Container {...restProps}>{children}</Container>;
}

Payment.Title = function PaymentTitle({children, ...restProps}){
    return <Title {...restProps}>{children}</Title>;
};


Payment.Button = function PaymentButton({children,...restProps}){
    return <Button {...restProps}>{children}</Button>;
};

Payment.SubTitle = function PaymentSunTitle({children,...restProps}){
    return <SubTitle {...restProps}>{children}</SubTitle>;
};

Payment.Col = function PaymenetCol({children, ...restProps}){
    return <Col {...restProps} >{children}</Col>;
};

Payment.PriceBox = function Pricebox({children,...restProps}){
    return <Price_box {...restProps}>{children}</Price_box>;
};

Payment.emphasis = function Emphasis({children,...restProps}){
    return <Emp {...restProps}>{children}</Emp>;
};

Payment.Head = function PHeader({children,...restProps}){
    return <Head {...restProps}>{children}</Head>
};