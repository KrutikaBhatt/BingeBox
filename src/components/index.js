export {default as Jumbotron} from './jumbotron';
export {default as Footer} from './Footer';
export {default as FAQs} from './FAQs';
export {default as OptingForm} from './OptForm';
export {default as Header} from './Header';
export {default as Feature} from './Feature';
export {default as Login_Form } from './Login_form';
export {default as PaymentComp } from './Payment';
export {default as PHeader} from './payment_header';