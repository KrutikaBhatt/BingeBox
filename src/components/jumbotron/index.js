import React from 'react';
import {Container,Inner,Pane,Title,subTitle,Image} from './styles/jumbotron';

export default function Jumbotron({ children,direction ='row', ...restProps}){
    return (
        <Inner direction={direction}>
           {children}
        </Inner>
    );
}

Jumbotron.Container = function JumbotronContainer({children,...restProps}){
    return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({children,...restProps}){
    return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({children,...restProps}){
    return <Title {...restProps}>{children}</Title>;
};

Jumbotron.subTitle = function Jumbotronsubtitle({children,...restProps}){
    return <subTitle {...restProps}>{children}</subTitle>;
};

Jumbotron.Image = function JumbotronImage({children,...restProps}){
    return <Image {...restProps} />;
};