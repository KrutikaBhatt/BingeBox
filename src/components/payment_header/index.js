import React from 'react';
import {Link as ReactRouterLink} from 'react-router-dom';
import {Background ,Container,Logo,ButtonLink} from './styles/Pheader';

export default function PHeader({bg =true,children,...restProps}){
    return bg ? <Background {...restProps}>{children}</Background> :children;
}

PHeader.Frame =function HeaderFrame({children,...restProps}){
    return(
        <Container {...restProps}>
            {children}
        </Container>
    );
};
PHeader.Logo  = function HeaderLogo({imageLink,...restProps}){
    return(
        <ReactRouterLink to={imageLink}>
            <Logo {...restProps}/>
        </ReactRouterLink>
    );
};

PHeader.ButtonLink = function HeaderButton({children , ...restProps}){
    return (<ButtonLink {...restProps}>{children}</ButtonLink>);
};
