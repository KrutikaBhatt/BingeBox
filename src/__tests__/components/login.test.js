import React from 'react';
import { render } from '@testing-library/react';
import { Login_Form } from '../../components';

jest.mock('react-router-dom');

describe('<Login_Form />', () => {
  it('renders the <Login_Form /> with populated data', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Login_Form>
        <Login_Form.Title>Sign In Now</Login_Form.Title>

        <Login_Form.Base>
          <Login_Form.Input placeholder="Email address" onChange={() => {}} />
          <Login_Form.Input type="password" placeholder="Password" onChange={() => {}} />
          <Login_Form.Submit type="submit" disabled>
            Sign In
          </Login_Form.Submit>
        </Login_Form.Base>

        <Login_Form.Text>
          New to Binge Box? <Login_Form.Link to="/signup">Sign up now.</Login_Form.Link>
        </Login_Form.Text>
        <Login_Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
        </Login_Form.TextSmall>
      </Login_Form>
    );

    expect(
      getByText("This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.")
    ).toBeTruthy();
    expect(getByText('Sign In Now')).toBeTruthy();
    expect(getByText('Sign In')).toBeTruthy();
    expect(getByText('Sign In').disabled).toBeTruthy();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Login_Form /> with an error', () => {
    const { container, getByText, queryByText } = render(
      <Login_Form>
        <Login_Form.Error>Your email address is already being used</Login_Form.Error>
        <Login_Form.Submit type="submit">Sign In</Login_Form.Submit>
      </Login_Form>
    );

    expect(getByText('Your email address is already being used')).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
