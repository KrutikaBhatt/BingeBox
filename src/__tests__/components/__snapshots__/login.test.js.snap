// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login_Form /> renders the <Login_Form /> with an error 1`] = `
<div
  class="form__Container-sc-1evz8eu-0 hwMbOK"
>
  <div
    class="form__Error-sc-1evz8eu-1 jqsEAX"
  >
    Your email address is already being used
  </div>
  <button
    class="form__Submit-sc-1evz8eu-8 jbcFPj"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<Login_Form /> renders the <Login_Form /> with populated data 1`] = `
<div
  class="form__Container-sc-1evz8eu-0 hwMbOK"
>
  <h1
    class="form__Title-sc-1evz8eu-3 gobude"
  >
    Sign In Now
  </h1>
  <p
    class="form__TextSmall-sc-1evz8eu-5 jopWPr"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </p>
</div>
`;
